packages/core/helper-plugin/lib/src/components/RelationInput/RelationInput.stories.mdx

```diff
@@ -0,0 +1,54 @@
+<!--- RelationInput.stories.mdx --->
+
+import { Meta, ArgsTable, Canvas, Story } from '@storybook/addon-docs';
+import { BaseLink } from '@strapi/design-system/BaseLink';
+import { Typography } from '@strapi/design-system/Typography';
+import { Badge } from '@strapi/design-system/Badge';
+import { Icon } from '@strapi/design-system/Icon';
+import Cross from '@strapi/icons/Cross';
+import { RelationList, RelationItem } from './index';
+import { relations } from './relationsDataMock';
+
+<Meta title="components/RelationInput" />
+
+# RelationInput
+
+WIP
+
+## Usage
+
+<Canvas>
+  <Story name="base">
+    {() => {
+      return (
+        <Main>
+          <RelationList overflow="top-bottom">
+            {relations.map(relation => {
+              const { title, status } = relation;
+              const variant = status === 'published' ? 'success' : 'secondary';
+              return (
+                <RelationItem endAction={<button type="button"><Icon width="12px" as={Cross} /></button>}>
+                  <BaseLink href="/">
+                    <Typography textColor="primary600">{title}</Typography>
```

Comment:
What do you think about moving `Typography` into `RelationItem`? It would make the component much easier to use and I think having `Badge` as a child of that wouldn't create additional problems ...

Reply:
Not sure I understand, if I add a title prop to handle Typography myself, then I BaseLink should be handled by RelationItem too?
If I remember well we agreed that only endAction would be a prop and the rest of it should be children?


```
export const RelationItem = ({ children, endAction, ...props }) => {
  return (
    <Flex
      paddingTop={3}
      paddingBottom={3}
      paddingLeft={4}
      paddingRight={4}
      hasRadius
      borderSize={1}
      borderColor="neutral200"
      justifyContent="space-between"
      {...props}
    >
      <Flex width="100%" justifyContent="space-between">
        <BaseLink href="/"><Typography>{title}</Typography></BaseLink> <=== TITLE HANDLING
        {children} <=== BADGE
      </Flex>
      {endAction && <Box paddingLeft={4}>{endAction}</Box>}
    </Flex>
  );
};
```

Reply:
Ah, right I remember. Mh, which leads to another question: which parts of the relation are clickable? I'd say best would be to have the whole surface clickable. Did you talk to @maevalienard about this?

I'm thinking that it doesn't make any sense to use `RelationItem` without the default `Typography` applied and that it would be nice, if we could encapsulate this is a bit more.

Wouldn't that work?

```js
<Flex>
  <Typography>
    <Flex width="100%" justifyContent="space-between">
      {children} /* BaseLink and Badge */
    </Flex>
  </Typography>
 
  {endAction && <Box paddingLeft={4}>{endAction}</Box>}
</Flex>
```

Reply:
mmh I'm not sure I'm a fan of wrapping a link + div (badge) inside a span, feels weird?
anyway I tried and it doesn't seem to be working well no - it applies the font but not the color + breaks UI

<img width="759" alt="image" src="https://user-images.githubusercontent.com/71838159/183677085-4775c890-271d-4f02-a238-74469c16bce2.png">

if we want the entire component to be clickable then indeed a lot needs to change
we would need to handle `BaseLink` inside `RelationItem` as we did for `FolderCard` or other components

Reply:
Can you check with @maevalienard what is actually wanted here, before we continue the discussion? 🙏🏼

Reply:
I'd say that the link is enough as a clickable area, especially because it's underlined, so we kind of expect this behavior :) I'm afraid the hit zones would be hard to deal with afterwards

Reply:
just as a note: the link is underlined because it's just a test - at the end I wanted to match the mockup so no underline!
do you prefer to keep the underline styling + only title clickable?

Reply:
Oh alright, thanks for letting me know. I think it's still fine though :)

Reply:
so @gu-stav - how would you prefer to handle the `Typography` thing then?
I agree that it would be nice not to have to add the `Typography` style by yourself when using the component but I still agree with our first decision:
having the entire content flexible by simply being `children` allows us to pass anything, maybe one day we'll need the text to be different, or maybe it will be different content - I think it's even more relevant if we move the component to the DS at some point.

Reply:
@gu-stav 👀😁

Reply:
@maevalienard I don't want to start a bikeshedding discussion here, but I think it would be very beneficial for a user and not to do it would be inconsistent with the rest of the DS:

- `Accordion` headers: full surface opens the panel
- `Button`: whole surface is clickable
- `Input`: the whole surface is interactive

... same for `Tabs`, `Combobox`es and `Select`. Even for `Option`s: they include the padding in the interactive area. I really think we should do the same here, because this lead to problems already for `Input` components.

I don't have an opinion on the `hover/focus` effect, but I think this discussion/ decision could also be decoupled from this :)

Reply:
@gu-stav to give my two cents I feel that the main interactivity the user will have with this component is adding & removing items, not clicking on the link.
if I try to click on the remove button and 1 time out of 2 I end up being redirected or having a popover/alert saying "careful are you sure you want to lose your progress", I'll be frustrated.

also I'm stuck since yesterday moving on with Relations until merge this PR. 
since it'll be a content-manager component at first before we later decide if we add it to the helper-plugin or the DS, I think we can easily iterate on it once we arrive close to finishing the feature, wdyt?

Reply:
Thanks for your feedback.

I think there is a difference between all these components. For the accordion, button, tab, combobox, select, the objective is usually to make an action within the page, so the hit zone should be big enough. Also, clicking on them doesn't have much of an impact. If you click, you could easily revert: the according is a collapse, you could switch tab, etc.
When it comes to this specific relation component, the objective is different, as Julie says you want to add relations and maybe removing them from the list, but having the whole area clickable would mean that you expect a proper action to be performed. However, here the click is related to navigation so the hit zone shouldn't be too big, otherwise the user could face too many misclick issues which is not the best in terms of experience to my opinion. 

Disclaimer: Here I'm not talking about development semantic but in terms of a random user experience who has no idea about the difference a button, a menu, a link, ... :) 

Happy to discuss furthermore in Slack if you are interested to keep this conversation going 😄

Reply:
I added Typography to RelationItem, ready for approve @gu-stav